e2e/nomostest/wait_for_sync.go

```diff
@@ -242,11 +243,8 @@ func (nt *NT) WaitForRootSyncSourceError(rsName, code string, message string, op
 			if err := nt.KubeClient.Get(rs.GetName(), rs.GetNamespace(), rs); err != nil {
 				return err
 			}
-			// Only validate the rendering status, not the Syncing condition
-			// TODO: Remove this hack once async sync status updates are fixed to reflect only the latest commit.
-			return testpredicates.ValidateError(rs.Status.Source.Errors, code, message, nil)
-			// syncingCondition := rootsync.GetCondition(rs.Status.Conditions, v1beta1.RootSyncSyncing)
-			// return validateRootSyncError(rs.Status.Source.Errors, syncingCondition, code, message, []v1beta1.ErrorSource{v1beta1.SourceError})
+			syncingCondition := rootsync.GetCondition(rs.Status.Conditions, v1beta1.RootSyncSyncing)
```

Comment:
Looks like these are in the WaitFor methods, not the newer preconditions we use with the WatchFor methods.
- Are these WaitFors still used? A lot? Can we replace them with watches?
- Do the WatchFor methods or their predicates have equivalent expectations?

Reply:
> Are these WaitFors still used? A lot? Can we replace them with watches?

Yes they are still used a lot. We could probably replace Wait with watches.

> Do the WatchFor methods or their predicates have equivalent expectations?

There seem to be `R*SyncHas*Error` predicate functions which have similar functionality, although those also don't verify the Syncing condition

Reply:
I'm frankly surprised this PR passes e2e tests. I have a feeling that it's still flaky, but we might need to replace them all with watches in order to enable debug diffs to figure out what the problem is. So I'm hesitant to merge this as-is, but I guess flaky failures is at least some signal... and we can always revert it again later.

What do you think?

Do you want to convert them all to watches first and add this condition there? That might give us more confidence.

Reply:
It occurs to me that if the retry timer is short or the resync timer is short, like we set it in the e2e tests, the Syncing condition may spend more time True than False due to it flipping back and forth before and after each apply attempt. We might need to lengthen the timers to make it less flaky.

Reply:
Haven't noticed any flakiness so far. Converted them all to use watches

Reply:
Cool. Thanks